src/openms/source/METADATA/AbsoluteQuantitationStandards.cpp

```diff
@@ -71,9 +71,11 @@ namespace OpenMS
       }
       for (const FeatureMap& fmap : feature_maps) // not all elements are necessarily processed (break; is present inside the loop)
       {
-        StringList sample_name;
-        fmap.getPrimaryMSRunPath(sample_name);
-        if (!sample_name.size() || sample_name[0] != run.sample_name) // if the FeatureMap doesn't have a sample_name, or if it is not the one we're looking for: skip.
+        StringList filename;
+        fmap.getPrimaryMSRunPath(filename);
+        String sample_name_1 = filename[0].substitute(".mzML", "");
```

Comment:
Global replace is dangerous? Maybe just check suffix?

Reply:
I like that.  Is there a way to easily suffix?